### PR TITLE
htop: merge both implementations

### PR DESCRIPTION
**Pull request by zimbatm**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

htop 2.0+ is now cross-platform.

This needs testing on OSX. IOKit might have to be re-introduced.

### CODE REVIEW

**Comment by nckx** [comment]
It would be nice to be able to ping @NixOS/osx-users...

**Comment by zimbatm** [comment]
Yes or having OSX builds happening

File: pkgs/tools/system/htop/default.nix [hunk 1]
```diff
@@ -13,9 +13,9 @@ stdenv.mkDerivation rec {
 
   meta = with stdenv.lib; {
     description = "An interactive process viewer for Linux";
-    homepage = http://htop.sourceforge.net;
+    homepage = https://hisham.hm/htop/;
     license = licenses.gpl2Plus;
-    platforms = platforms.linux;
+    platforms = platforms.all;
```
Comment (fpletz) [inline pkgs/tools/system/htop/default.nix]: The website states:
> This release includes code supporting Linux, FreeBSD, OpenBSD and Mac OS X.

Comment (zimbatm) [inline pkgs/tools/system/htop/default.nix]: Too late ^_^. Fixed in master as 57dd725e853b54c578e3b73f520756be3ff2359c

**Comment by zimbatm** [comment]
@jagajaga did you test it on OSX ?

**Comment by jagajaga** [comment]
@zimbatm I supposed https://github.com/NixOS/nixpkgs/pull/14065#issuecomment-198817252 meant that you've tested..

**Comment by zimbatm** [comment]
Ah sorry, it was a misunderstanding then.

**Pull request merged**